k2/csrc/array.h

```diff
@@ -312,12 +312,24 @@ class Array1 {
    */
   Array1 operator[](const Array1<int32_t> &indexes) const {
     NVTX_RANGE(K2_FUNC);
-    const ContextPtr &c = Context();
-    K2_CHECK(c->IsCompatible(*indexes.Context()));
+    const ContextPtr &c = GetContext(*this, indexes);
```

Comment:
We cannot use a reference here. `GetContext` returns a variable that lives on the stack.